src/main/scala/org/allenai/ike/IkeBatchAPI.scala

```diff
@@ -0,0 +1,226 @@
+    *                         because you want the filename and the sentence/content for this pattern's match
+    * @return A Seq of ApiExtractionResult case classes which hold the results
+    */
+  def doSearch(patternName: String, req: SearchRequest, indexWithContent: String): Seq[ApiExtractionResult] = {
+    import org.allenai.ike.JsonSerialization._
+
+    val query = SearchApp.parse(req).get
+
+    val (tables, patterns) = req.userEmail match {
+      case Some(userEmail) => (
+        Tablestore.tables(userEmail),
+        Tablestore.namedPatterns(userEmail)
+      )
+      case None => (Map.empty[String, Table], Map.empty[String, NamedPattern])
+    }
+
+    val interpolatedQuery = QueryLanguage.interpolateQuery(
+      query,
+      tables,
+      patterns,
+      word2vecPhrasesSearcher,
+      tableExpander
+    ).get
+
+    val foundResults = searchApps.map { case (name, searcher) => searcher.search(interpolatedQuery, req.config).get }
+    val qexpr = SearchApp.parse(req).get
+
+    // brakes the flatmap if empty
+    if (foundResults.head.isEmpty) {
+      return List[ApiExtractionResult]()
+    }
+
+    val extractions = foundResults.flatMap {
+      case rr =>
+        rr.map {
+          case blab =>
+            val docId = blab.sourceDocument.getOrElse(-1) // TODO: Should barf on -1
+
+            val extractedString = blab.matchWords.mkString(" ")
+
+            val extractionStart = blab.matchOffset.start
+            val extractionEnd = blab.matchOffset.end
+
+            // Lucene document is XML. Assuming one index where this documnent lives for now. TODO: sketchy...
+            val contentXML = XML.loadString(searchApps(indexWithContent).searcher.getContent(docId))
+            // get the words out, so we can make them into a string. We want the whole sentence, so need this
+            val sourceString = contentXML.text
+
+            // Again, assuming one index where this document lives for now... TODO: sketchy...
+            val sourceArticle = searchApps(indexWithContent).searcher.document(docId).get("fromInputFile").toString
+
+            val apiEx = ApiExtractionResult(patternName, extractedString, sourceString,
+              sourceArticle, extractionStart, extractionEnd)
+            Set(apiEx)
+        }.reduce(_ ++ _) // append them all together
+    }
+    extractions.toSeq
+  }
+}
+
+// example usage... call from command line as: "run-main org.allenai.ike.IkeCommandlineRunner"
+object IkeCommandlineRunner extends App {
+  val ikeCmd = new IkeBatchAPI()
+  val extracts = ikeCmd.extract("myconfig.conf")
+  extracts.foreach {
+    case result =>
+      println(result)
+      println(result.patternName)
+      println(result.extraction)
+      println(result.sourceString)
+      println(result.sourceArticle)
+      println(result.exractionStartTokenLoc + " " + result.extractionEndTokenLoc)
+      println("--------")
```

Comment:
No prob, great idea. I was mostly including this for debug and example of usage, but do you think it's actually useful from command line?

For the time being, I can just make it output some JSON or something.